linkerd/admin/src/main/scala/com/twitter/finagle/client/buoyant/ClientStateHandler.scala

```diff
@@ -0,0 +1,41 @@
+package com.twitter.finagle.client.buoyant
+
+import com.twitter.finagle.{Addr, Address, Path, Service}
+import com.twitter.finagle.client.ClientRegistry
+import com.twitter.finagle.http.{MediaType, Request, Response}
+import com.twitter.finagle.loadbalancer.LoadBalancerFactory
+import com.twitter.util.{Future, Try}
+import io.buoyant.config.Parser
+
+class ClientStateHandler extends Service[Request, Response] {
+
+  private case class ClientState(state: String, addresses: Seq[String] = Nil)
+
+  private[this] val mapper = Parser.jsonObjectMapper(Nil)
+
+  override def apply(request: Request): Future[Response] = {
+    val entries = ClientRegistry.registrants
+      .filter { entry =>
+        // only display clients with a Path name
+        Try(Path.read(entry.addr)).isReturn
+      }.map { entry =>
+        val LoadBalancerFactory.Dest(va) = entry.params[LoadBalancerFactory.Dest]
```

Comment:
Are we confident that `LoadBalancerFactory.Dest`'s Var is reference-counted?  Do we need to ensure this higher up in the stack when we get the result from an interpreter?